src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -347,6 +360,7 @@ private int FindEntry(TKey key)
                             // Test in if to drop range check for following array access
                             if ((uint)i >= (uint)entries.Length || (entries[i].hashCode == hashCode && EqualityComparer<TKey>.Default.Equals(entries[i].key, key)))
                             {
+                                value = ref entries[i].value;
```

Comment:
Seems suspicious with the bounds check in the if statement immediately above. Perhaps this break doesn't necessarily mean that a match was found?

Reply:
Ah, of course! Have been staring at this for a while and not seeing the wood for the trees.